vspace/src/main/webapp/WEB-INF/views/staff/spaces/space.html

```diff
@@ -7,503 +7,445 @@
+                    }
+                    setLinkPosition(moduleLink,link,false,false,false);
+                    $("#main-space").append(moduleLink);
+                    $('[data-link-id="'+link.link.id+'"]').css('cursor', 'pointer');
+                    $('[data-link-id="'+link.link.id+'"]').click(function(e) {
+                        var linkData = $('#modulelinkId-'+link.link.id).data();
+                        makeModuleLinksEditable(linkData["linklabel"], link.link.id,linkData["linkrotation"],linkData["linktarget"],link.positionX,link.positionY,link.id, linkData["linktype"]);
+                    });
+                }
+            })
+            externalLinks.forEach(function(link,index){
+                if(link != null){
+                    var modalExternalLink ;
+                    if(link.type=="IMAGE" && link.image != null){
+                        var modalExternalImageURL = [[@{/api/image/}]]+link.image.id;
+                        modalExternalLink = $('<img id="externalModallinkId-'+link.externalLink.id+'"  data-linkTarget="'+link.link.module.id+'" data-linkType="'+link.type+'" data-linkLabel="'+link.externalLink.name+'" class="externalLink-"'+link.externalLink.id+'" data-link-id="'+link.externalLink.id+'" src="'+modalExternalImageURL+'" />');
+                    }
+                    else{
+                        modalExternalLink = $('<span id="externalModallinkId-'+link.externalLink.id+'" data-link-id="'+link.externalLink.id+'" data-linkTarget="'+link.externalLink.externalLink+'" data-linkType="'+link.type+'" data-linkLabel="'+link.externalLink.name+'" class="externalLink-'+link.externalLink.id+' Info_cz_Class"><svg class="Ellipse_8_c"><ellipse fill="rgba(222,222,222,1)" class="Ellipse_8_c_Class" rx="14.5" ry="14.5" cx="14.5" cy="14.5"></ellipse></svg><svg class="Ellipse_10_c"><ellipse fill="rgba(240,240,240,1)" class="Ellipse_10_c_Class" rx="12.5" ry="12.5" cx="12.5" cy="12.5"></ellipse></svg><svg class="Ellipse_9_c"><ellipse fill="rgba(255,255,255,1)" class="Ellipse_9_c_Class" rx="10.5" ry="10.5" cx="10.5" cy="10.5"></ellipse></svg><span class="icon-ext-link Icon_awesome_info_staff_e"></span><p class="label-'+link.externalLink.id+'" data-link-id="'+link.externalLink.id+'"><span class="tooltiptext">'+link.externalLink.name+'</span></p></span>'); 
+                    }
+                    setLinkPosition(modalExternalLink,link,false,true,true);
+                    $("#space").append(modalExternalLink);
+                    $('[data-link-id="'+link.externalLink.id+'"]').css('cursor', 'pointer');
+                    $('[data-link-id="'+link.externalLink.id+'"]').click(function(e) {
+                        var linkData = $('#externalModallinkId-'+link.externalLink.id).data();
+                        makeExternalLinksEditable(linkData["linklabel"],link.externalLink.id,linkData["linktarget"],link.positionX,link.positionY,link.id,linkData["linktype"]);
+                    });
+                }
+            })

```

Comment:
the above three loops seem to have a lot of duplicate code. extract common code into functions, pass the parts that are different (e.g. ids, attribute prefixes, etc.).